src/components/dashboard/subsettings/TokenSettings.tsx

```diff
@@ -120,25 +167,59 @@ export default function TokenSettings({ id }: Props) {
             </option>
           </SelectV2>
           <AnimatePresence>
-            {token.gateway && <CustomGatewayWarning />}
+            {token?.gateway && <CustomGatewayWarning />}
           </AnimatePresence>
         </div>
       )}
-      <ButtonV2 fullWidth onClick={() => removeToken(id)}>
-        <TrashIcon style={{ marginRight: "5px" }} />
-        {browser.i18n.getMessage("remove_token")}
-      </ButtonV2>
+      <ButtonWrapper>
+        {isAoToken && (
+          <ButtonV2
```

Comment:
Adding loading spinner to the Refresh Token will help to know if refresh is in progress.